### DISCUSSION TITLE
Downloading files from MinIO distributed server results in proxy downloads from other servers

### DISCUSSION THREAD

**Discussion by faryar76** [minio/minio#17127] [Q&A] [answered]
I am currently using MinIO in distributed mode with 4 servers, arranged in two racks, with two servers in each rack. While downloading a file from server 1, I noticed that around 50% of the file is downloaded from server 3, which is located in a different rack. I would prefer that the download only occurs from the server that I am requesting the download from, or at least from a server in the same rack.

I have tried setting the region configuration option, but it does not seem to affect which server is chosen for the proxy download. Is there any way to configure MinIO to prioritize servers in the same rack, or to disable this behavior entirely?

Any insights or suggestions on how to resolve this issue would be greatly appreciated. Thank you in advance for your help!

**Comment by harshavardhana** [comment, marked as answer]
This is not an issue this is erasure coding at play, our implementation provides rack level tolerance. This is very much an important feature from HA point of view.

**Comment by faryar76** [comment]
Thanks you 🙏
